source/diagnostics.tex

```diff
@@ -2491,12 +2494,13 @@
 \impldef{default semantics of \tcode{is_debugger_present}}.
 
 \begin{note}
-When tracing the execution of a program with a debugger, an implementation
-returns \tcode{true}. An implementation performs an immediate query, as needed,
-to determine if the program is traced by a debugger. On Windows or equivalent
-systems, this can be achieved by calling the \tcode{::IsDebuggerPresent()} Win32
-function. On POSIX, this can be achieved by checking for a tracer parent process,
-with best effort determination that such a tracer parent process is a debugger.
+It is intended that, using an immediate (uncached) query to determine if the
+program is being traced by a debugger, an implementation returns \tcode{true}
+only when tracing the execution of the program with a debugger. On Windows or
+equivalent systems, this can be achieved by calling the
+\tcode{::IsDebuggerPresent()} Win32 function. For systems compatible with
+ISO/IEC 23360:2021, this can be achieved by checking for a tracing process, with
```

Comment:
This replaces "POSIX" with "Linux Standard Base".  Is there a particular reason for that?  Does POSIX not cover the needed functionality?

Reply:
Although `ptrace` is present in most UNIX-like systems, it isn't part of POSIX. The LSB is a much narrower scope than POSIX, but it _is_ a standard, unlike families like "BSD" or "SysV". So if we want a standard that includes `ptrace`, LSB makes sense.

However, the wording never mentions `prrace` it just talks about "tracing" and "a tracer parent process" which is still meaningful even in e.g. HP-UX which replaced `ptrace` with `ttrace`, and in Solaris through `ptrace` but also via procfs. So maybe it's OK.

Reply:
> However, the wording never mentions `prrace` it just talks about "tracing" and "a tracer parent process" which is still meaningful even in e.g. HP-UX which replaced `ptrace` with `ttrace`, and in Solaris through `ptrace` but also via procfs. So maybe it's OK.

Using "POSIX" in conjunction with the handwaving implies that the handwaving is necessarily meaningful for all possible implementations of POSIX. That it is probably meaningful for extant implementations of POSIX does not make me feel any better about using "POSIX" without qualification.